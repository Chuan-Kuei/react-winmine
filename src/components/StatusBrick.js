import React from "react";
import PropTypes from "prop-types";

import "./statucBrick.css";

const StatusBrick = ({
  status,
  onClick: handleClick,
  onContextMenu: handleRightClick
}) => {
  return (
    <div
      styleName={`container ${status}`}
      onClick={handleClick}
      onContextMenu={handleRightClick}
    />
  );
};

StatusBrick.defaultProps = {
  status: "smile",
  onContextMenu: e => {
    e.preventDefault();
  }
};
StatusBrick.propTypes = {
  status: PropTypes.oneOf(["smile", "winner", "lost"]),
  onClick: PropTypes.func.isRequired,
  onContextMenu: PropTypes.func
};

export default StatusBrick;
