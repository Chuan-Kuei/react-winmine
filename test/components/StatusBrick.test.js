import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";

import StatusBrick from "../../src/components/StatusBrick";

describe("StatusBrick", () => {
  const defaultOnclick = () => {};
  test("render correctly", () => {
    const wrapper = renderer
      .create(<StatusBrick onClick={defaultOnclick} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test("props status is smile", () => {
    const wrapper = shallow(
      <StatusBrick status="smile" onClick={defaultOnclick} />
    );
    expect(wrapper.hasClass("container")).toBe(true);
    expect(wrapper.hasClass("smile")).toBe(true);
  });

  test("props status is lost", () => {
    const wrapper = shallow(
      <StatusBrick status="lost" onClick={defaultOnclick} />
    );
    expect(wrapper.hasClass("container")).toBe(true);
    expect(wrapper.hasClass("lost")).toBe(true);
  });

  test("props status is lost", () => {
    const wrapper = shallow(
      <StatusBrick status="lost" onClick={defaultOnclick} />
    );
    expect(wrapper.hasClass("container")).toBe(true);
    expect(wrapper.hasClass("lost")).toBe(true);
  });

  test("right click is preventDefault", () => {
    const wrapper = shallow(<StatusBrick onClick={defaultOnclick} />);
    const mockEvent = {
      preventDefault: jest.fn()
    };
    wrapper.find(".container").simulate("contextMenu", mockEvent);
    expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
  });

  test("left click, click one times, and capture 1 time click", () => {
    const mockOnClick = jest.fn();
    const wrapper = shallow(<StatusBrick onClick={mockOnClick} />);
    wrapper.find(".container").simulate("click");
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
