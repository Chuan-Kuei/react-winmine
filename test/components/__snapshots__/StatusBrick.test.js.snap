// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBrick render correctly 1`] = `
<div
  className="container smile"
  onClick={[Function]}
  onContextMenu={[Function]}
/>
`;
